### ISSUE TITLE
Manifest keys unsupported in the current manifest version should be reported as warnings

### ISSUE THREAD

**Issue by willdurand** [mozilla/addons-linter#3843]
When a MV2 add-on contains MV3 keys, the submission on AMO should not fail. Instead, we should have warning messages. Same behavior with a MV3 add-on contains MV2 keys.

**Comment by willdurand** [comment]
@rpl feel free to update the issue with more information

**Comment by AlexandraMoga** [comment]
@willdurand I've enabled the `enable-mv3-submissions` switch on stage and tried to validate add-ons in the following formats:

1. MV2 with MV3 specific keys:
```
{
  "manifest_version": 2,
  "name": "manifest v3",
  "description": "testing v2 and v3 manifest early",
  "version": "1.0.0",
  "background": {	
    "service_worker": "background.js"
  }
```
Validation results:
![image](https://user-images.githubusercontent.com/31961530/128192468-e2c9b368-bbfa-4e6f-af8b-c1aa75489aa1.png)

2. MV3 with MV2 specific keys:
```
{
  "manifest_version": 3,
  "name": "DevHub-Listed-Ext-05.13",
  "description": "Listed extension for release testing",
  "version": "10.195",
  "background": {
    "scripts": [
      "background_script.js"
    ]
  }
```
Validation results:
![image](https://user-images.githubusercontent.com/31961530/128193088-6dd12d5b-8058-45d2-8291-ac9e30682e45.png)

I'm still receiving errors instead of warnings for `MANIFEST_FIELD_UNSUPPORTED`. I've also tried validating these addons in the CLI with the latest linter release and I'm still getting errors instead of warnings.

**Comment by willdurand** [comment]
@AlexandraMoga what happens with the waffle switch disabled or with the CLI without any manifest version option? (i.e. it should only validate MV2)

**Comment by willdurand** [comment]
Also, after having talked to @rpl a bit, `background` is a valid key in MV2 so we are kinda strict about the sub-properties, which explains why there are errors for the background sub-props.

**Comment by AlexandraMoga** [comment]
@willdurand with the switch disabled or `--max-manifest-version=3` unspecified in the CLI, the MV2 with MV3 specific keys produces the same validation results (as above example 1). 
MV3 with MV2 specific keys will trigger the manifest not supported error, plus the additional errors/warnings seen before:
```
ERRORS:

Code                      Message                    Description                                      File            Line   Column
JSON_INVALID              "/manifest_version"        Your JSON file could not be parsed.
                          should be <= 2
MANIFEST_FIELD_REQUIRED   "/background/service_wo…   See https://mzl.la/1ZOhoEN (MDN Docs) for more   manifest.json
                          rker" is a required        information.
                          property
WARNINGS:

Code                         Message                   Description                                      File            Line   Column
MANIFEST_FIELD_UNSUPPORTED   "/browser_action" is in   "/browser_action" is in a format not supported   manifest.json
                             a format not supported    in manifest versions > 2.
                             in manifest versions >
                             2.
```

**Comment by willdurand** [comment]
I think that is expected. Maybe try with another prop like `host_permissions` (which is MV3 only)

**Comment by AlexandraMoga** [comment]
@willdurand I'm not sure about `host_permissions`. If I add them to either an MV2 or MV3 manifest, then validate the extension with and without `--max-manifest-version=3` there is no related error or warning in the validation results. The validation results are exactly as before.

This is how I've added `host_permissions` in the manifest: 

```
  "host_permissions": [
	"http://www.blogger.com/*",
	"http://*.google.com/*"
  ]
```

**Comment by willdurand** [comment]
Yeah, that's good. This was a problem in FF last week. If this isn't a problem with the linter, we're good.
